### PR TITLE
change the flash memory region reallocation for spl boot

### PR DESCRIPTION
**Pull request by jianchenadi**
* 0x000000 - 0x010000 : SPL	(64KiB)
 * 0x010000 - 0x012000 : ENV	(8KiB)
 * 0x012000 - 0x092000 : U-Boot (512KiB)
 * 0x092000 - 0x0D2000 : DTB	(256KiB)
 * 0x0D2000 - 0x5D2000 : ARGS	(5MB)
 * 0x5D2000 - 0x19D2000: Linux Kernel - zImage (20MB)
 * 0x19D2000 - 0x4000000: Linux Kernel - initramfs (38MB)

Signed-off-by: jian chen <jian.chen@analog.com>

### CODE REVIEW

File: include/configs/sc589-mini.h [hunk 1]
```diff
@@ -241,11 +242,14 @@
...
-#define CONFIG_SYS_SPI_KERNEL_SIZE	0x1460000 /* 20MB */
-#define CONFIG_SYS_SPI_INITRAM_OFFS	0x1500000 /* CONFIG_SYS_SPI_KERNEL_OFFS + CONFIG_SYS_SPI_KERNEL_SIZE */
+#define CONFIG_SYS_SPI_U_BOOT_SIZE  0x80000    /* 512KiB */
+#define CONFIG_SYS_SPI_DTB_OFFS		0x92000    /* CONFIG_SYS_SPI_U_BOOT_OFFS + CONFIG_SYS_SPI_U_BOOT_SIZE */
+#define CONFIG_SYS_SPI_DTB_SIZE		0x40000    /* 256KiB */
+#define CONFIG_SYS_SPI_ARGS_OFFS	0xD2000    /* CONFIG_SYS_SPI_DTB_OFFS + CONFIG_SYS_SPI_DTB_SIZE */
+#define CONFIG_SYS_SPI_ARGS_SIZE	0x500000  /* 5MB */
+#define CONFIG_SYS_SPI_KERNEL_OFFS	0x5D2000  /* CONFIG_SYS_SPI_ARGS_OFFS + CONFIG_SYS_SPI_ARGS_SIZE */
+#define CONFIG_SYS_SPI_KERNEL_SIZE	0x1400000  /* 20MB */
+#define CONFIG_SYS_SPI_INITRAM_OFFS	0x19D2000  /* CONFIG_SYS_SPI_KERNEL_OFFS + CONFIG_SYS_SPI_KERNEL_SIZE */
```
Comment (davidgibsonadi) [inline include/configs/sc589-mini.h:252]: What are the current size of the uboot, dtb, args, kernel and initram sizes for the actual binaries, compared to the space we are allocating? How much growth space is there in each of these areas?

Comment (jianchenadi) [reply include/configs/sc589-mini.h:252]: Current size:
u-boot-spl.ldr: 24KB
u-boot.bin: 272KB
uImage: 2.3MB
dtb: 19KB
initram: 2.8MB
args: 5MB reserved

**Comment by sunnyfengadi** [comment]
splupdate, kupdate, splexport can work well, the u-boot can test pass, so ship it

**Review by sunnyfengadi — APPROVED** [review]

File: include/configs/sc589-mini.h (hunk 1 above)
Comment (davidgibsonadi) [reply include/configs/sc589-mini.h:252]: I think the declared sizes are good, the current demo is significantly stripped back. These sizes should mean that users don't need to resize when their app grows. But we need to be aware that we are reserving a lot of memory.
Let's make sure that this is documented on the wiki.

**Pull request merged**